scripts/get_e2e_artifacts.py

```diff
@@ -0,0 +1,162 @@
+
+
+def get_test_paths_and_names(test_suite_path: str):
+  """Get the paths Bazel stores test outputs in and the matching test names."""
+  targets = utils.get_test_targets(test_suite_path)
+  test_paths = [_target_to_testlogs_path(target) for target in targets]
+  test_names = [
+      _target_to_test_name(target, test_suite_path) for target in targets
+  ]
+  return test_paths, test_names
+
+
+def check_collision(filename: str, test_name: str, written_paths: Set[str],
+                    paths_to_tests: Dict[str, str]):
+  """Check that we aren't overwriting files unless we expect to."""
+  # Note: We can't use a check that the files have identical contents because
+  # tf_input.mlir can have random numbers appended to its function names.
+  # See https://github.com/google/iree/issues/3375
+
+  expected_collision = any([name in filename for name in EXPECTED_COLLISIONS])
+  if filename in written_paths and not expected_collision:
+    raise ValueError(f'Collision found on {filename} between {test_name}.py '
+                     f'and {paths_to_tests[filename]}.py')
+  else:
+    written_paths.add(filename)
+    paths_to_tests[filename] = test_name
+
+
+def update_path(archive_path: str):
+  """Update the --input_file flag with the new location of the compiled.vmfb"""
+  backend_path = archive_path.split('traces')[0]  # 'ModuleName/backend_name'.
+  compiled_path = os.path.join(FLAGS.artifacts_dir, backend_path,
+                               'compiled.vmfb')
+  flagfile_path = os.path.join(FLAGS.artifacts_dir, archive_path)
+  for line in fileinput.input(files=[flagfile_path], inplace=True):
+    if line.strip().startswith('--input_file'):
+      print(f'--input_file={compiled_path}\n', end='')
+    else:
+      print(line, end='')
+
+
+def extract_artifacts(test_path: str, test_name: str, written_paths: Set[str],
+                      paths_to_tests: Dict[str, str]):
+  """Unzips all of the benchmarking artifacts for a given test and backend."""
+  outputs = os.path.join(test_path, 'test.outputs', 'outputs.zip')
+  archive = zipfile.ZipFile(outputs)
+  # Filter out directory names.
+  filenames = [name for name in archive.namelist() if name[-1] != os.sep]
+
+  for filename in filenames:
+    # Check for collisions.
+    check_collision(filename, test_name, written_paths, paths_to_tests)
+
+    # Extract and update flagfile path.
+    if not FLAGS.dry_run:
+      archive.extract(filename, FLAGS.artifacts_dir)
+      if filename.endswith('flagfile'):
+        update_path(filename)
+
+
+def main(argv):
+  del argv  # Unused.
+
+  # Convert test suite shorthands to full test suite targets.
+  test_suites = [SUITE_NAME_TO_TARGET[suite] for suite in FLAGS.test_suites]
+
+  written_paths = set()
+  paths_to_tests = dict()
+
+  for test_suite in test_suites:
+    if FLAGS.run_test_suites and not FLAGS.dry_run:
+      subprocess.check_call([
+          'bazel', 'test', test_suite, '--color=yes',
```

Comment:
Ok GitHub is absolutely refusing to let me respond to the existing thread :angry:, but what I mean by stream is stream stdout so that a user sees that things are progressing rather than it just hanging forever.

Reply:
Oh, it already did/does that! With the `--color=yes` flag it looks just as if you'd called it from the shell.